pam/pam-module-transaction-dummy.go

```diff
@@ -0,0 +1,295 @@
+package main
+
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+)
+
+// PamModuleTransactionDummy is an implementation of PamModuleTransaction for
+// testing purposes.
+type PamModuleTransactionDummy struct {
+	Items       map[pam.Item]string
+	Env         map[string]string
+	Data        map[string]any
+	convHandler pam.ConversationHandler
+}
+
+// NewPamModuleTransactionDummy returns a new PamModuleTransactionDummy.
+func NewPamModuleTransactionDummy(convHandler pam.ConversationHandler) pam.ModuleTransaction {
+	return &PamModuleTransactionDummy{
+		convHandler: convHandler,
+	}
+}
+
+// InvokeHandler is called by the C code to invoke the proper handler.
+func (m *PamModuleTransactionDummy) InvokeHandler(handler pam.ModuleHandlerFunc,
+	flags pam.Flags, args []string) (pam.Status, error) {
+	return pam.Abort, pam.Abort
+}
+
+// SetItem sets a PAM information item.
+func (m *PamModuleTransactionDummy) SetItem(item pam.Item, value string) error {
+	if m.Items == nil {
+		m.Items = make(map[pam.Item]string)
+	}
+
+	m.Items[item] = value
+	return nil
+}
+
+// GetItem retrieves a PAM information item.
+func (m *PamModuleTransactionDummy) GetItem(item pam.Item) (string, error) {
+	if m.Items == nil {
+		return "", nil
+	}
+	return m.Items[item], nil
```

Comment:
You can just:
```Go
return m.Items[item], nil
```

You can't assign to a nil map in Go, but you can try to access a key from it. If it doesn't exist, an empty value for the type will be returned (in this case, it would `return "", nil`)

Reply:
Oh right... I even used this in some cases in go-pam, missed this one :)